services/ipqsservicesample/service/ipqsclient.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+import (
+  "github.com/gin-gonic/gin"
+  "github.com/gin-contrib/cors"
+  "net/http"
+  "io"
+  "fmt"
+  "encoding/json"
+)
+
+type PathBindings struct {
+  Email string `uri:"email"`
+  Phone string `uri:"phone"`
+  URL string `uri:"url"`
+  IP string `uri:"ip"`
+}
+
+type APIKey struct {
+  Key string `json:"key"`
+}
+
+const apiBase string = "https://www.ipqualityscore.com/api/json"
+var apiKey string
+
+func executeRequest(url string) (response []byte, err error) {
+  resp, err := http.Get(url)
+  if err != nil {
+    return nil, err 
+  }
+  
+  defer resp.Body.Close()
+  
+  body, err := io.ReadAll(resp.Body)
+  if err != nil {
+    return nil, err
+  }
+  
+  return body, nil
+}
+
+func main() {
+  router := gin.Default()
+  
+  config := cors.DefaultConfig()
+  config.AllowAllOrigins = true
+  
+  router.Use(cors.New(config))
```

Comment:
One thing that'd be nice to address that may have some CORS-related configs to adjust: It'd be great if our service routes could always have flexibility for trailing '/' characters. I'll have to confirm again to see what the current behavior is, but when playing with things originally, I saw issues with redirects. Let's explore that more and see if we can make things better + more flexible.